src/Utilities/ProcessExtensions.cs

```diff
@@ -77,10 +77,16 @@ private static void GetAllChildIdsUnix(int parentId, ISet<int> children)
 
         private static void KillProcessUnix(int processId)
         {
-            RunProcessAndWaitForExit(
-                "kill",
-                $"-TERM {processId}",
-                out string _);
+            try
+            {
+                using Process process = Process.GetProcessById(processId);
+                process.Kill();
+            }
+            catch (ArgumentException)
```

Comment:
I think it's InvalidOperationException not ArgumentException: [https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill?view=netframework-4.8](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill?view=netframework-4.8)

Reply:
It's both. `GetProcessById` throws `ArgumentException` and `Kill` throws `InvalidOperationException`.

From .NET Core 3 onward, `Kill` no longer throws `InvalidOperationException` when the process has already exited.